dockerfiles/debian11.Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM debian:bullseye as debian11-builder
+
+RUN apt update && apt install -y \
+    build-essential git tree software-properties-common dirmngr apt-transport-https ufw libpcre2-dev zlib1g-dev
```

Comment:
It is easier to manage using individual dockerfiles rather than a single large dockerfile. Also, we no longer use a download of zlib or pcre. Instead we use the same techniques as the new OAuth Proxy module.